test/unit/controllers/ApplicationControllerSpec.scala

```diff
@@ -878,6 +878,287 @@ class ApplicationControllerSpec extends UnitSpec with MockitoSugar with WithFake
       }
     }
 
+    "manageTeamMembers" when {
+      "the user is a superuser" should {
+        "show 200 OK" in new Setup {
+          givenASuccessfulSuperUserLogin()
+          givenTheAppWillBeReturned()
+
+          val result = await(addToken(underTest.manageTeamMembers(applicationId))(aSuperUserLoggedInRequest))
+
+          status(result) shouldBe OK
+        }
+      }
+
+      "the user is not a superuser" should {
+        "show 401 Unauthorized" in new Setup {
+          givenASuccessfulLogin()
+          givenTheAppWillBeReturned()
+
+          val result = await(addToken(underTest.manageTeamMembers(applicationId))(aLoggedInRequest))
+
+          status(result) shouldBe UNAUTHORIZED
+        }
+      }
+    }
+
+    "addTeamMember" when {
+      "then user is a superuser" should {
+        "show 200 OK" in new Setup {
+          givenASuccessfulSuperUserLogin()
+          givenTheAppWillBeReturned()
+
+          val result = await(addToken(underTest.addTeamMember(applicationId))(aSuperUserLoggedInRequest))
+
+          status(result) shouldBe OK
+        }
+      }
+
+      "then user is not a superuser" should {
+        "show 401 Unauthorixed" in new Setup {
+          givenASuccessfulLogin()
+          givenTheAppWillBeReturned()
+
+          val result = await(addToken(underTest.addTeamMember(applicationId))(aLoggedInRequest))
+
+          status(result) shouldBe UNAUTHORIZED
+        }
+      }
+    }
+
+    "addTeamMemberAction" when {
+      "the user is a superuser" when {
+        "the form is valid" should {
+          val email = "email@example.com"
+          val role = "DEVELOPER"
+
+          "call the service to add the team member" in new Setup {
+            givenASuccessfulSuperUserLogin()
+            givenTheAppWillBeReturned()
+
+            given(mockApplicationService.addTeamMember(any[Application], any[Collaborator], anyString)(any[HeaderCarrier])).willReturn(Future.successful(ApplicationUpdateSuccessResult))
+
+
+            val request = aSuperUserLoggedInRequest.withFormUrlEncodedBody(("email", email), ("role", role))
+
+            val result = await(addToken(underTest.addTeamMemberAction(applicationId))(request))
+
+            verify(mockApplicationService).addTeamMember(eqTo(application.application), eqTo(Collaborator(email, CollaboratorRole.DEVELOPER)), eqTo("superUserName"))(any[HeaderCarrier])
+          }
+
+          "redirect back to manageTeamMembers when the service call is successful" in new Setup {
```

Comment:
I think this might be broken if, due to a previous comment, an error is thrown instead